presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -389,6 +389,7 @@ public Visitor(Optional<StageExecutionDescriptor> stageExecutionStrategy, TypePr
         public Void visitExplainAnalyze(ExplainAnalyzeNode node, Void context)
         {
             addNode(node, "ExplainAnalyze");
+            addNode(node.getSources().get(0), "ExchangeNode");
```

Comment:
This is not the right place for this fix. Instead, we should add an unconditional exchange node in AddExchanges.visitExplainAnalyze. Specifically, we should remove the check that does:

```
if ((child.getNode() instanceof ExchangeNode) && ((ExchangeNode) child.getNode()).getType() == ExchangeNode.Type.GATHER) {
    return rebaseAndDeriveProperties(node, child);
}
```